mycpp/gc_str.cc

```diff
@@ -482,3 +488,96 @@ List<Str*>* Str::split(Str* sep) {
 
   return result;
 }
+
+// TODO: We should reduce the number of allocations and copies done here.
+static inline Str* _StrFromFormat(const char* fmt, va_list args) {
+  int n = strlen(fmt);
+  auto beg = std::cregex_iterator(fmt, fmt + n, gStrFmtRegex);
+  auto end = std::cregex_iterator();
+  mylib::BufWriter* w = Alloc<mylib::BufWriter>();
```

Comment:
can be `auto*`

Although actually how about using `std::string`  for a growable string, and then create a string at the end?


BufWriter will create GC pressure when it grows

Also then you don't have to write StrFromC() fragments

I suppose you have one extra copy at the end (since BufWriter::getvalue() no longer copies) but I think std::string is still better, since we know the lifetime

Reply:
Yeah, I'd really like to avoid unnecessary GC pressure here. I'll adapt to use `std::string`